lib/rjgit_adapter/git_layer_rjgit.rb

```diff
@@ -160,11 +160,14 @@ def grep(query, options={})
           blobs << RJGit::Blob.new(@git.jrepo, item[:mode], item[:path], walk.lookup_blob(ObjectId.from_string(item[:id]))) if item[:type] == 'blob'
         end
         result = []
+        regex_query = query.scan(/"([^"]+)"|(\S+)/).flatten.compact.map {|word| Regexp.escape(word)}
```

Comment:
The variable name `regex_query` is not really fitting here at this point since it is still only an array of search terms. `search_terms` would be better.

Also, instead of `word` for the scoped variable I would prefer `term` here.